client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -506,55 +506,64 @@ def build_template(
+                 workfile for an initial version. It will skip saving if
+                 a version already exists.
+            workfile_creation_enabled (bool): Whether the call is part of
+                creating a new workfile.
+                When True, we only build if the current file is not
+                an existing saved workfile but a "new" file. Basically when
+                enabled we assume the user tries to load it only into a
+                "New File" (unsaved empty workfile).
+                When False, the default value, we assume we explicitly want to
+                build the template in our current scene regardless of current
+                scene state.
 
         """
-        if any(
-            value is None
-            for value in [
-                template_path,
-                keep_placeholders,
-                create_first_version,
-            ]
-        ):
-            template_preset = self.get_template_preset()
-            if template_path is None:
-                template_path = template_preset["path"]
-            if keep_placeholders is None:
-                keep_placeholders = template_preset["keep_placeholder"]
-            if create_first_version is None:
-                create_first_version = template_preset["create_first_version"]
 
-        # check if first version is created
-        created_version_workfile = False
-        if create_first_version:
-            created_version_workfile = self.create_first_workfile_version()
-
-        # if first version is created, import template
-        # and populate placeholders
+        # Get default values if not provided
         if (
-            create_first_version
-            and workfile_creation_enabled
-            and created_version_workfile
+            template_path is None
+            or keep_placeholders is None
+            or create_first_version is None
         ):
+            preset = self.get_template_preset()
+            template_path: str = template_path or preset["path"]
+            if keep_placeholders is None:
+                keep_placeholders: bool = preset["keep_placeholder"]
+            if create_first_version is None:
+                create_first_version: bool = preset["create_first_version"]
+
+        # Build the template if we are explicitly requesting it or if it's
+        # an unsaved "new file".
+        is_new_file = not self.host.get_current_workfile()
+        explicit_build_requested = not workfile_creation_enabled
+        if is_new_file or explicit_build_requested:
+            self.log.info(f"Building the workfile template: {template_path}")
             self.import_template(template_path)
             self.populate_scene_placeholders(
                 level_limit, keep_placeholders)
 
-            # save workfile after template is populated
-            self.save_workfile(created_version_workfile)
+        if not workfile_creation_enabled:
```

Comment:
```suggestion
        if explicit_build_requested:
```
?

Reply:
I had thought of that - but I felt the "skipping of saving" felt WAY weirder if it'd skip it just because an explicit build was requested so I figured that keeping it as it was is actually clearer in this if statement. What do you think? Happy to change it though.

Reply:
not sure I follow, `not workfile_creation_enabled == explicit_build_requested` no? To me it's much easier to read conditionals without negation clauses so I personally find it better but up to you

Reply:
yeah I see what you mean... but isn't the other conditional the same then?

Reply:
It's the same value as `explicit_build_requested` yes, but it just made more sense to "skip saving" `if not workfile_creation_enabled`. Anyway, I'll just refactor your suggestion because there's also an argument to make for the `not` in that if statement making it harder to read.

Reply:
what about `disable_workfile_creation` ?

Reply:
No matter how we rename the argument or the methods - it requires updating the dependent addons that call this function. Hence, I'd make that a separate PR. And, if we're going down that route I'd really do what I described [here](https://github.com/ynput/ayon-core/pull/871#issuecomment-2338251853). Building a template should not need to care at all whether it's a new file or not - it should just load template and build regardless. If we do need certain logic related to "build_and_save_on_new_file" or whatever then we should separate that out. It's really polluting this function (and its arguments) which should just be:

```python
builder.build_template()
```

Also, we have "full control" in a DCC on deciding when to 'build on new file' so I'd even argue we don't need to check then whether current file is saved - but just respond to the _new file_ callback and then have it initialize accordingly. Which means this is just as sensible:

```python
def on_new():
    builder.build_template()
    core.save_first_file()
```

Especially because we also mentioned that it may also make sense to allow saving that "first workfile" even if not template was used but we just want to initialize the workfile name for the artist potentially, etc.